client/src/main/java/io/pravega/client/stream/impl/ReaderGroupState.java

```diff
@@ -519,7 +519,7 @@ void update(ReaderGroupState state) {
                     Entry<Segment, Long> entry = iter.next();
                     Segment segment = entry.getKey();
                     Long offset;
-                    if (ownedSegments == null) {
+                    if (ownedSegments == null || ownedSegments.isEmpty()) {
```

Comment:
Two questions:

1- In which cases do we expect it to be null and which cases we expect it to be empty? I'm wondering, for example, whether `ownedSegments` being `null` is actually an error. If so, we should consider having a precondition. 
2- There are a bunch of `null` checks in this file alone. I'm wondering if any of those should be checking for empty set instead. Did you check?

Reply:
>whether ownedSegments being null is actually an error.

ownedSegments is null in the example shown in the integration test when reader is made offline with position object as null.

>  I'm wondering if any of those should be checking for empty set instead. Did you check?

No, I have not yet checked it.